spec/index.adoc

```diff
@@ -560,7 +560,15 @@ In order to call a canister, the user makes a POST request to `/api/v2/canister/
 * `method_name` (`text`): Name of the canister method to call
 * `arg` (`blob`): Argument to pass to the canister method
 
-The HTTP response to this request has an empty body and HTTP status 202, or a HTTP error (4xx or 5xx). Paranoid agents should not trust this response, and use <<http-read-state,`read_state`>> to determine the status of the call.
+The HTTP response to this request can have the following responses:
+
+* 202 HTTP status with empty body. Implying the request was accepted by the IC for further processing. Users should use <<http-read-state,`read_state`>> to determine the status of the call.
+* 200 HTTP status with non-empty body. Implying an execution pre-processing error occurred. The body of the response contains more information about the IC specific error encountered. The body is a CBOR map with the following fields:
+** `reject_code` (`nat`): The reject code (see <<reject-codes>>).
+** `reject_message` (`text`): a textual diagnostic message.
+** `error_code` (`text`): an optional implementation-specific textual error code (see <<error-codes>>).
+* 4xx HTTP status for client errors (e.g. malformed request or canister not found). Retrying the request will likely have the same outcome.
```

Comment:
The suggestion seems too generic: "doing X might fail".

If anything, I'd rather mention 429 explicitly. Or say something along the lines of "The client should not retry the request without modification" (in some cases, said modification would be to e.g. use a different URL or authenticate; in other cases, it would simply mean "wait until the conditions are right").

Reply:
I mentioned 429 explicitly now. I did not go for the sentence "The client should not retry the request without modification" because with 429 you can make the very same request again.